src/br/unicamp/meca/mind/MecaMind.java

```diff
@@ -277,140 +293,189 @@ private void mountMotivationalCodelets() {
+					
+					if(perceptualCodelets != null) {
+						for(PerceptualCodelet perceptualCodelet : perceptualCodelets) {
+							if(perceptualCodelet != null && perceptualCodelet.getId() != null) {
+								ArrayList<String> perceptualCodeletsIds = behaviorCodelet.getPerceptualCodeletsIds();
+								if(perceptualCodeletsIds != null) {
+									for(String perceptualCodeletId : perceptualCodeletsIds) {
+										if(perceptualCodeletId != null && perceptualCodelet.getId().equalsIgnoreCase(perceptualCodeletId)) {
+											behaviorCodelet.addInputs(perceptualCodelet.getOutputs());
 										}
 									}
 								}
 							}
 						}
 					}
 
-					if (soarCodelet != null && soarCodelet.getId() != null && motivationalBehavioralCodelet.getSoarCodeletId() != null) {
-						if (soarCodelet.getId().equalsIgnoreCase(motivationalBehavioralCodelet.getSoarCodeletId())) {
-							motivationalBehavioralCodelet.addBroadcasts(soarCodelet.getOutputs());
+					if (soarCodelet != null && soarCodelet.getId() != null && behaviorCodelet.getSoarCodeletId() != null) {
+						if (soarCodelet.getId().equalsIgnoreCase(behaviorCodelet.getSoarCodeletId())) {
+							behaviorCodelet.addBroadcasts(soarCodelet.getOutputs());
 						}
 
 					}
 
-					insertCodelet(motivationalBehavioralCodelet);
+					insertCodelet(behaviorCodelet);
 
 				}
 			}
 		}
 	}
+	
+	private void mountActionFromPerceptionCodelets() {
+		if (actionFromPerceptionCodelets != null) {
+			for (ActionFromPerception actionCodelet : actionFromPerceptionCodelets) {
+				if (actionCodelet != null && actionCodelet.getId() != null
+						&& actionCodelet.getPerceptualCodeletsIds() != null
+						&& actionCodelet.getMotivationalCodeletsIds() != null
+						&& actionCodelet.getMotorCodeletId() != null) {
 
-	private void mountReactiveBehavioralCodelets() {
-		if (getReactiveBehavioralCodelets() != null) {
-			for (ReactiveBehavioralCodelet reactiveBehavioralCodelet : getReactiveBehavioralCodelets()) {
-				if (reactiveBehavioralCodelet != null && reactiveBehavioralCodelet.getId() != null
-						&& reactiveBehavioralCodelet.getPerceptualCodeletsIds() != null
-						&& reactiveBehavioralCodelet.getMotorCodeletId() != null) {
-
-					insertCodelet(reactiveBehavioralCodelet);
+					insertCodelet(actionCodelet);
 					/*
 					 * Outputs
 					 */
 					if (motorCodelets != null) {
 						for (MotorCodelet motorCodelet : motorCodelets) {
 							if (motorCodelet != null && motorCodelet.getId() != null) {
 								if (motorCodelet.getId()
-										.equalsIgnoreCase(reactiveBehavioralCodelet.getMotorCodeletId())) {
-									reactiveBehavioralCodelet.addOutputs(motorCodelet.getInputs());
+										.equalsIgnoreCase(actionCodelet.getMotorCodeletId())) {
+									actionCodelet.addOutputs(motorCodelet.getInputs());
 								}
 							}
 						}
 					}
 					/*
 					 * Inputs
 					 */
+					
+					if (motivationalCodelets != null) {
+						for (MotivationalCodelet motivationalCodelet : motivationalCodelets) {
```

Comment:
motivationalCodelets.stream().forEach(........

Reply:
Nice suggestion, indeed, but as is just a matter of style, resulting in the same end, will let it to a future PR.